### ISSUE TITLE
NaN loss when training

### ISSUE THREAD

**Issue by HITyycc** [salesforce/TabularSemanticParsing#22]
Hi, 
when loss became 'nan', it stopped at pdb.set_trace():

(Pdb) l
697                 return 0
698             loss = F.nll_loss(masked_inputs, masked_targets)
699             if torch.isnan(loss):
700                 import pdb
701                 pdb.set_trace()
702  ->         if loss > 1e8:
703                 import pdb
704                 pdb.set_trace()
705             return loss

And the Model output is nan.

(Pdb) p masked_inputs
tensor([[     nan,      nan,      nan,      nan,      nan,      nan,      nan,
              nan,      nan,      nan,      nan,      nan,      nan,      nan,
              nan,      nan,      nan,      nan,      nan,      nan,      nan,
              nan,      nan,      nan,      nan,      nan,      nan,      nan,

Why Nan loss? How to deal with Nan loss?

**Comment by elaruwan** [comment]
I've also had this vanishing gradient when trying to apply automatic mixed preicision APEX/amp to the model.
